File: soal-uts-1/soal-uts-1.py
class PolaFor:
    def printPattern(self, baris):
        print("Banyak Baris : %d\n" % (baris))
        for i in range(baris, 0, -1):
            print("%d. " % (baris-i+1), end='')
            for j in range(baris, i, -1):
                print("  ", end='')
            for k in range(1, i+1):
                print("* ", end='')
            print("\n", end='')

class PolaWhile:
    def printPattern(self, baris):
        i = baris
        print("Banyak Baris : %d\n" % (baris), end='')
        while i >= 1 :
            j = baris
            k = 1
            print("%d. " % (baris-i+1), end='')
            while j > i:
                print("  ", end='')
                j -= 1
            while k <= i:
                print("* ", end='')
                k += 1
            print("\n", end='')
            i -= 1

if __name__ == "__main__":
    coba = PolaFor()
    coba1 = PolaWhile()

    print("==========Pola For==========")
    coba.printPattern(4)
    
    print("==========Pola While==========")
    coba1.printPattern(4)

File: soal-uts-1/soal-uts-1.cpp
#include <stdio.h>

class PolaFor{
    public:
        void printPattern(int baris){
            printf("Banyak Baris : %d\n", baris);
            for(int i= baris; i >= 1; i--){
                printf("%d. ", baris-i+1);
                for(int j=baris; j > i;j--){
                    printf("  ");
                }  
                for(int k=1;k<=i;k++){
                    printf("* ");
                }  
                printf("\n");
            }
        }
};

class PolaWhile{
public:
    void printPattern(int baris){
        int i = baris;
        int j, k;
        printf("Banyak Baris : %d\n", baris);
        while(i >= 1) {
            j = baris;
            k = 1;
            printf("%d. ", baris-i+1);
            while(j > i) {
                printf("  ");
                j--;
            }
            while(k <= i) {
                printf("* ");
                k++;
            }
            printf("\n");
            i--;
        }
    }
};

int main(int argc, char const *argv[])
{
    PolaFor coba = PolaFor();
    PolaWhile coba1 = PolaWhile();

    printf("==========Pola For==========\n");
    coba.printPattern(4);

    printf("==========Pola While==========\n");
    coba1.printPattern(4);
    return 0;
}

